db/cache/charts.go

```diff
@@ -917,34 +956,29 @@ func (charts *ChartData) Chart(chartID, binString, axisString string) ([]byte, e
 	return data, nil
 }
 
-// Keys used for the chartResponse data sets.
-var responseKeys = []string{"x", "y", "z"}
-
-// Encode the slices. The set lengths are truncated to the smallest of the
-// arguments.
-func (charts *ChartData) encode(sets ...lengther) ([]byte, error) {
+// Encode the data sets. Optionally add arbitrary additional data as part of the
+// chartRespnse seed. A nil seed is allowed.
+func encode(sets lengtherMap, seed chartResponse) ([]byte, error) {
 	if len(sets) == 0 {
 		return nil, fmt.Errorf("encode called without arguments")
 	}
-	smaller := sets[0].Length()
-	for _, x := range sets {
-		l := x.Length()
-		if l < smaller {
+	smaller := -1
+	for _, set := range sets {
+		l := set.Length()
+		if smaller == -1 {
+			smaller = l
+		} else if l < smaller {
 			smaller = l
 		}
 	}
-	response := make(chartResponse)
-	for i := range sets {
-		rk := responseKeys[i%len(responseKeys)]
-		// If the length of the responseKeys array has been exceeded, add a integer
-		// suffix to the response key. The key progression is x, y, z, x1, y1, z1,
-		// x2, ...
-		if i >= len(responseKeys) {
-			rk += strconv.Itoa(i / len(responseKeys))
-		}
-		response[rk] = sets[i].Truncate(smaller)
+	if len(seed) == 0 {
+		seed = make(chartResponse)
```

Comment:
A micro-optimization I might suggest is `make(chartResponse, len(sets))` because it would also serve to convey intent.   Very minor thing though.